te_dp/src/te_tcp_dp.c

```diff
@@ -713,15 +703,10 @@ int curl_socket_timer_cb(CURLM *multi, long timeout_ms, void *userp)
     * It calls on_session_uv_timeout()
     */
     te_session_t *session = (te_session_t *)userp;
-    if (unlikely(is_session_config_state_stopped(session->session_cfg_p))) {
-       return 0;
-    }
     if (timeout_ms < 0) {
        uv_timer_stop(&session->tcp->cm_timer);
     }
     else {
-        if(timeout_ms == 0)
```

Comment:
Seems like this code was directly inherited from https://curl.se/libcurl/c/multi-uv.html. We are on fast path, there is no place to dilly-dally. This change induced a 1 ms delay in the receive path